### PR TITLE
Removal of python-future compatibility code 

### PR DESCRIPTION
**Pull request by knabar**
This PR implements issue #528 to review the OMERO.web source code to update all paths with either Python 2 or Python 2/3 logic to pure Python 3+, similar to @sbesson's work for OMERO.py at https://github.com/ome/omero-py/pull/390.

### CODE REVIEW

**Comment by knabar** [comment]
@sbesson I have gotten to the point where tests are now failing because the underlying `omero-py` is not Python 3.12 compatible, so we have to coordinate somehow

File: tox.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tox]
 # Default tox environment when run without `-e`
-envlist = py38, py39, py310, py311
```
Comment (sbesson) [inline tox.ini:3]: Is the plan to drop support for Python 3.8 in the next OMERO.web release

Comment (knabar) [reply tox.ini:3]: Don't think so, that is my mistake - I'll add py38 back in

**Review by sbesson — APPROVED** [review]
With the caveat that the temporary commit pointing to my branch should be updated before merging, all changes look sensible to me from an initial code review. I'll let the OME team schedule the functional testing of this set of maintenance/Python 3.12 support PR.

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
[2edac67](https://github.com/ome/omero-web/pull/531/commits/2edac67bd76fd9f07cbefc572532f82450c0047c) broke today's installation of omero-server. It highlights an issue in the ci-build

**Comment by sbesson** [comment]
I think the issue is rather with 52c84a6dec8529c2c9df796efbd544f737ff9f14 rather than the `tox.ini` change which is unused by OME CI as far as I know. Here's the failing workflow as I understand it:

- [OMERO-python-superbuild-build](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-build/444/) creates distributions of the `omero-web` and `omero-py` packages including all included PRs
- [OMERO-server](https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-server/400/console) tries to install these packages in order. When installing `omero-web`, the `install_requires` modifies 52c84a6dec8529c2c9df796efbd544f737ff9f14 checks out my `omero-py` branch from GitHub and runs `setup.py`
- unlike the artifacts from the Jenkins job above, the Git checkout does not include a `target` folder and `omero-py` tries to download `blitz-python` - see https://github.com/ome/omero-py/blob/1ca2f8729e684c5e702931ac4c7b2f768d950d0e/setup.py#L60-L64
- the `JENKINS_URL` is misconfigured on https://merge-ci.openmicroscopy.org/jenkins causing the job to fail.

I suspect force pushing away 52c84a6dec8529c2c9df796efbd544f737ff9f14 should be sufficient to keep testing this PR through GitHub actions (as per the `tox.ini` workflow) and meet the requirement of the OME Jenkins pipeline.

**Comment by jburel** [comment]
I have now removed the excluded label since I found the issue in the CI job leading to the unexpected behaviour

**Comment by jburel** [comment]
Sorry yes, wrong cut and paste.
Problem is sorted now

**Comment by jburel** [comment]
The ``JENKINS_URL`` has been reviewed in https://github.com/ome/devspace/pull/212/commits/1859719ce12c7071925fc7ac763fae036f4a0d62

**Comment by jburel** [comment]
See https://github.com/ome/devspace/pull/212/commits/096b024b077a737522b7ea304b44dd5be6601466 for the adjustment (new CI)
Similar changes made to merge-ci https://merge-ci.openmicroscopy.org/jenkins/job/OMERO-server/402/

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#2](https://BRANCH-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/2/). See the [console output](https://BRANCH-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/2/consoleText) for more details.
Possible conflicts:
  - PR #17 will-moore 'Right panel performance fixes'
    - omeroweb/webclient/views.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#452](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/452/). See the [console output](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/452/consoleText) for more details.
Possible conflicts:
  - PR #17 will-moore 'Right panel performance fixes'
    - omeroweb/webclient/views.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#3](https://BRANCH-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/3/). See the [console output](https://BRANCH-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/3/consoleText) for more details.
Possible conflicts:
  - PR #17 will-moore 'Right panel performance fixes'
    - omeroweb/webclient/views.py

--conflicts

**Comment by snoopycrimecop** [comment]
Conflicting PR. Removed from build [OMERO-python-superbuild-push#453](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/453/). See the [console output](https://snoopycrimecop-ci.openmicroscopy.org/jenkins/job/OMERO-python-superbuild-push/453/consoleText) for more details.
Possible conflicts:
  - PR #17 will-moore 'Right panel performance fixes'
    - omeroweb/webclient/views.py

--conflicts

**Comment by knabar** [comment]
> What is the reason for dropping Python 3.8 from the testing matrix?

Just an oversight, I made a mistake undoing the removal.

File: omeroweb/webgateway/marshal.py [hunk 2]
```diff
@@ -296,8 +295,8 @@ def set_if(k, v, func=lambda a: a is not None):
 
     def decode(shape_field):
```
Comment (jburel) [inline omeroweb/webgateway/marshal.py:296]: The comment line 382 could probably be removed now
```
# Handle string for python2 and bytes python3. TODO: lower level fix
```

File: omeroweb/custom_forms.py [hunk 3]
```diff
@@ -62,12 +61,10 @@ def full_clean(self):
...
-                    if (
-                        value is not None
-                        and isinstance(value, basestring)
-                        and len(value) > 0
-                    ):
-                        value = str(smart_str(value))
+                    if value is not None and isinstance(value, str) and len(value) > 0:
+                        value = str(
+                            smart_str(value)
+                        )  # TODO: This is probably a noop now
```
Comment (will-moore) [inline omeroweb/custom_forms.py:67]: if `isinstance(value, str)` then we don't need to do anything - can `pass` here?

**Review by will-moore — APPROVED** [review]
Looks good, thanks.
Just one TODO to cleanup but not a blocker 👍

File: omeroweb/custom_forms.py (hunk 3 above)
Comment (knabar) [reply omeroweb/custom_forms.py:67]: I probably left it in because I wasn't sure that `smart_str` would not do something

**Pull request merged**